Cannot build for `GOARCH=386`

**What steps did you take:**

Build for, say, `windows/386`:

```
GOOS=windows GOARCH=386 go build ./...
```

**What happened:**

Int overflow. 🤖 💀 

```
# carvel.dev/imgpkg/pkg/imgpkg/registry
pkg/imgpkg/registry/registry.go:202:36: cannot use 9999999999 (untyped int constant) as int value in argument to rand.Intn (overflows)
```

**What did you expect:**

It builds w/o problems.

**Anything else you would like to add:**

`imgpkg` is not distributed for `*/386`. That's not something we need to change. However, since `carvel.dev/imgpkg` can be consumed as a library, it should be compatible.

**Environment:**

- imgpkg version (use `imgpkg --version`): n/a
- Docker registry used (e.g. `Docker HUB`): n/a
- OS: `windows/386`

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

--- a/pkg/imgpkg/registry/registry.go
+++ b/pkg/imgpkg/registry/registry.go
@@ -199,7 +199,7 @@ func NewSimpleRegistryWithTransport(opts Opts, rTripper http.RoundTripper) (*Sim
 
 	sessionID := opts.SessionID
 	if sessionID == "" {
-		sessionID = fmt.Sprint(rand.Intn(9999999999))
+		sessionID = fmt.Sprint(rand.Int31())
 	}
 	baseRoundTripper = NewImgpkgRoundTripper(baseRoundTripper, sessionID)
 

